util/chunk/chunk.go

```diff
@@ -30,24 +32,60 @@ type Chunk struct {
 	// numVirtualRows indicates the number of virtual rows, which have zero column.
 	// It is used only when this Chunk doesn't hold any data, i.e. "len(columns)==0".
 	numVirtualRows int
+	// capacity indicates the max number of rows that the chunk wants to hold.
+	// The accurate row count after executor.Next is guaranteed by executor implement.
+	// It is used only during executor's execution, and grow by `Renew` or `Reset`.
+	capacity int
 }
 
 // Capacity constants.
 const (
 	InitialCapacity = 32
+	varElemLen      = -1
 )
 
 // NewChunkWithCapacity creates a new chunk with field types and capacity.
 func NewChunkWithCapacity(fields []*types.FieldType, cap int) *Chunk {
+	return New(fields, cap, cap) //FIX ME in continue PR, this keep every thing same.
+}
+
+// New creates a new chunk.
+//  cap: the max number of rows that the chunk wants to hold.
+//  capLimit: the max limit for max number of rows.
+func New(fields []*types.FieldType, cap, capLimit int) *Chunk {
```

Comment:
1. the meaning of `cap` and `capLimit` makes people confused...`
2. seems `capLimit` is not used in this function

Reply:
2. I make a bug in previous, address comment orz..., this funcation will check `cap` not over `capLimit`
1. having any good advise~~?

Reply:
yes, for current plan, it will be equal to `tidb_max_chunk_size` just like previous does.